core/service/slot.ts

```diff
@@ -62,37 +67,31 @@ class SlotService {
         return slot * config.CONSTANTS.FORGING.SLOT_INTERVAL;
     }
 
-    /**
-     * @method
-     * @return {number} current slot number + 1.
-     */
-    public getNextSlot(): number {
-        const slot = this.getSlotNumber();
-
-        return slot + 1;
-    }
-
-    /**
-     * @method
-     * @param {number} nextSlot
-     * @return {number} input next slot + delegates.
-     * // todo mb drop it ?
-     */
-    public getLastSlot(nextSlot: number): number {
-        return nextSlot + config.CONSTANTS.ACTIVE_DELEGATES;
-    }
-
-    public roundTime(date: Date): number {
-        return Math.floor(date.getTime() / SECOND) * SECOND;
+    public getSlotRealTime(slot: number): number {
+        const slotTime = slot * config.CONSTANTS.FORGING.SLOT_INTERVAL;
+        return this.getRealTime(slotTime);
     }
-
-    public getTheFirsSlot() {
-        return Math.floor(this.getSlotNumber() / config.CONSTANTS.ACTIVE_DELEGATES) * config.CONSTANTS.ACTIVE_DELEGATES;
+    
+    public generate(blockId: string, delegates: Array<Delegate>, firstSlot: Slot): Slots {
```

Comment:
maybe should rename to `generateSlots`, because it could be understood as `generateSlot`